Add endpoint capability to lock type

Zitrium31 opened the pull request:
Purpose of this change:
- add endpoint support to lock type in `homeassistant.ts`
- ~~add test for home assistant discovery config for such device with made of several endpoints using `state` key in `toZigbee` converter : _onOff, windowCovering, lock~~_

In short, this change supplements issue Koenkk/zigbee2mqtt#24352 for lock.

~~Prerequisite : Added tests required the dummy garage door and lock device model `GADOLOCK` in `custom_devices_diy` definitions with [PR #8529 in ZHC repo](https://github.com/Koenkk/zigbee-herdsman-converters/pull/8529).~~

🙂  Thanks in advance for the review of this PR. Should you have any remark, please ask.

Zitrium31 committed:
Add endpoint support to lock type

Zitrium31 committed:
Add discovery test for lock device with endpoint
Require device 'Garage door with lock (GADOLOCK)'
added in zhc:src/devices/custom_devices_diy.ts with PR
https://github.com/Koenkk/zigbee-herdsman-converters/pull/8529

Zitrium31 committed:
Merge branch 'dev' into feat-lock-multiendpoint

Nerivec commented:
This should be implemented for an actual device, not a dummy one.
- Don't want dummy devices bloating converters
- Don't want a crafted dummy device to direct features testing since that could be different from an actual device
- _If no actual device uses this, no point in having it_

Zitrium31 committed:
Merge branch 'Koenkk:master' into feat-lock-multiendpoint

Zitrium31 committed:
Revert "Add discovery test for lock device with endpoint"
This device does not fulfill zhc policy because it does not exist in real life.

This reverts commit 13a5f7cdd7ae1e08fa14f07f1a2fc094a09d288c.

Zitrium31 commented on lib/extension/homeassistant.ts:
```diff
@@ -747,22 +747,22 @@ export default class HomeAssistant extends Extension {
                 const discoveryEntry: DiscoveryEntry = {
                     type: 'lock',
-                    object_id: 'lock',
+                    object_id: endpoint ? `lock_${endpoint}` : 'lock',
```
Any advice how to pass test coverage on `endpoint ? lock_${endpoint}` case ?

Koenkk replied:
You can add a ` /* v8 ignore next */` above it.

Zitrium31 committed:
Ignore coverage for lock with endpoint
No lock with endpoint available at the time being
Tested locally with a DIY lock device made of several endpoints

Koenkk committed:
Merge branch 'dev' into feat-lock-multiendpoint

Koenkk commented:
Thanks!
